components/builder-api/src/server/mod.rs

```diff
@@ -58,12 +58,15 @@ impl HttpGateway for ApiSrv {
 
     fn router(config: Arc<Self::Config>) -> Router {
         let basic = Authenticated::new(&*config);
+        let mut opt = basic.clone();
```

Comment:
It might be a bit cleaner to have the 'optional' call return self, then you could do:
let mut opt = basic.clone().optional();

Reply:
Thinking about this more - we might actually want a different middleware type, eg, OptionallyAuthenticated. The normal Authenticated use in the code can always do an unwrap to get the session.  With optional in the mix, there's always a question of whether that unwrap is going to be safe or not.  So a different type that makes this explicit might be better.